File: internal/model/advance_allocation.go
package model

import (
	"time"

	"github.com/google/uuid"
	"github.com/shopspring/decimal"
)

type AdvanceAllocation struct {
	ID              uuid.UUID       `db:"id"`
	TenantID        uuid.UUID       `db:"tenant_id"`
	AdvanceID       uuid.UUID       `db:"advance_id"`
	AdvanceType     string          `db:"advance_type"`
	TargetID        uuid.UUID       `db:"target_id"`
	TargetType      string          `db:"target_type"`
	AllocatedAmount decimal.Decimal `db:"allocated_amount"`
	AllocationDate  time.Time       `db:"allocation_date"`
	VoucherID       *uuid.UUID      `db:"voucher_id"`
	VoucherNo       *string         `db:"voucher_no"`
	Remark          *string         `db:"remark"`
	CreatedBy       *uuid.UUID      `db:"created_by"`
	CreatedAt       time.Time       `db:"created_at"`
}

File: frontend/src/api/modules/advance_allocation.ts
import request from '@/api/request'
import type { ApiResponse } from '@/types/api'

export interface AdvanceAllocation {
  id: string
  advance_id: string
  advance_type: 'receipt' | 'payment'
  target_id: string
  target_type: 'ar' | 'ap'
  allocated_amount: string
  allocation_date: string
  voucher_no?: string
  remark?: string
  created_at: string
}

export interface AllocatePayload {
  advance_id: string
  advance_type: 'receipt' | 'payment'
  target_id: string
  target_type: 'ar' | 'ap'
  allocated_amount: string
  allocation_date: string
  remark?: string
}

export function allocateAdvance(payload: AllocatePayload): Promise<ApiResponse<AdvanceAllocation>> {
  return request.post('/advance-allocations', payload)
}

export function autoMatchAdvance(id: string): Promise<ApiResponse<AdvanceAllocation[]>> {
  return request.post(`/advance-allocations/${id}/auto-match`)
}

export function listAdvanceAllocations(advanceId: string): Promise<ApiResponse<{ list: AdvanceAllocation[]; total: number }>> {
  return request.get('/advance-allocations', { params: { advance_id: advanceId } })
}

export function listAdvanceAllocationsByTarget(targetId: string): Promise<ApiResponse<{ list: AdvanceAllocation[]; total: number }>> {
  return request.get('/advance-allocations/by-target', { params: { target_id: targetId } })
}
